File: 240929/가장 짧은 부분합/shortest-subtotal.py
n, s = map(int, input().split())
arr = list(map(int, input().split()))
leng = len(arr)

j = 0
p_sum = 0
ans = -1
for i in range(leng):
    while j < leng:
        if p_sum >= s:
            ans = j - i if ans == -1 else min(ans, j-i)
            # print(i, ans)
            break
        p_sum += arr[j]
        j += 1
    p_sum -= arr[i]

print(ans)

File: 240929/가장 짧은 부분합/shortest-subtotal.java
import java.util.*;

public class Main {
    public static final int INT_MAX = Integer.MAX_VALUE;
    public static final int MAX_N = 100000;

    public static int[] arr = new int[MAX_N + 1];
    public static int n, s;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        n = sc.nextInt();
        s = sc.nextInt();
        for(int i = 1; i <= n; i++)
            arr[i] = sc.nextInt();

        int j = 0, pSum = 0, ans = INT_MAX;
        for(int i = 1; i <= n; i++) {
            while(j + 1 <= n && pSum < s) {
                pSum += arr[j + 1];
                j++;
            }

            if (pSum < s) continue;


            ans = Math.min(ans, j - i + 1);
            pSum -= arr[i];
        }

        System.out.print(ans);
    }
}